src/classes/GiftBatch.cls

```diff
@@ -93,35 +66,29 @@ public inherited sharing class GiftBatch {
     }
 
     @TestVisible
-    private GiftBatch(Gifts gifts) {
-        this.gifts = gifts;
+    private GiftBatch() {}
+
+    public GiftBatch(GiftBatchId giftBatchId) {
+        this.giftBatchId = giftBatchId;
+        this.batch = giftBatchSelector.selectGiftBatchBy(this.giftBatchId);
+        this.giftTemplate = new GiftTemplate(this.batch.Form_Template__r.Template_JSON__c);
+        this.gifts = new Gifts(this.giftBatchId.value(), new GiftSelector(giftTemplate));
     }
 
     public GiftBatch(Id batchId) {
-        this.batchId = batchId;
-        this.batch = giftBatchSelector.selectGiftBatchById(this.batchId);
+        this.giftBatchId = new GiftBatchId(batchId);
+        this.batch = giftBatchSelector.selectGiftBatchBy(this.giftBatchId);
+        this.giftTemplate = new GiftTemplate(this.batch.Form_Template__r.Template_JSON__c);
+        this.gifts = new Gifts(this.giftBatchId, new GiftSelector(giftTemplate));
     }
 
     public Boolean hasElevateWidget() {
```

Comment:
I'm a little surprised to see this in the domain class, though I certainly wouldn't dig my heels in over this. It seems like the presence of the elevate widget is more of a "view-like" behavior?

Reply:
You're absolutely right! This doesn't quite belong here. We need to clean this out once Gift Entry has been completely decoupled from the legacy BGE classes.

This is in place currently because a legacy BGE utility class needs to check the batch's template for the elevate widget. It needs to do that so that we can display a couple of Elevate-related Data Import fields in the BGE datatable.

I'm hoping we can perform this removal during the Gift Entry refactor work item next sprint.

Reply:
Taking another look at this... I actually did deliver this method in the new GiftTemplate class as well. What do you think about replacing the `GiftBatch.hasElevateWidget()` method with a `GiftBatch.template()`?

We could then update the legacy BGE_BatchGiftEntryUtil to do something like `giftBatch.template().hasElevateWidget()`.